README.md

```diff
@@ -1,16 +1,1483 @@
+        "bytes": "7aEDQMyGqYw2iwP7uIn+Kav5AWe9l5VnL72Gpkzs1Azp+zs6vnixQPa1aCSrok4XwKkhSlFRmRN8YbyohB6iDFl4CQ8"
+      }
+    }
+  }
+}
+```
+
+### 5.3.3 Causal Invocations
+
+```json
+{
+  "bafyreievhy7rnzot7mnzbnqtiajhxx7fyn7y2wkjtuzwtmnflty3767dny": {
+    "on": "dns:example.com?TYPE=TXT",
+    "call": "crud/update",
+    "input": {
+      "value": "hello world"
+    }
+  },
+  "bafyreiemzjyz5sxmeatnlykpzr2feribz6yyxjlvvufsrbodx43t4wfi5m": {
+    "v": "0.1.0",
+    "run": {
+      "/": "bafyreievhy7rnzot7mnzbnqtiajhxx7fyn7y2wkjtuzwtmnflty3767dny"
+    },
+    "auth": {
+      "/": "bafyreia6tgroy43hkzywlg4ks7zxyvzgc62wifdvqjyy6uo3hrrrehgb3u"
+    },
+    "cause": {
+      "/": "bafyreibytdkxnyy4zzkiznz4psosqdhazizjrllzmu373ej4r6bckcvwju"
+    },
+    "prf": [
+      {
+        "/": "bafyreieynwqrabzdhgl652ftsk4mlphcj3bxchkj2aw5eb6dc2wxieilau"
+      }
+    ]
+  },
+  "bafyreia6tgroy43hkzywlg4ks7zxyvzgc62wifdvqjyy6uo3hrrrehgb3u": {
+    "scope": [
+      {
+        "/": "bafyreievhy7rnzot7mnzbnqtiajhxx7fyn7y2wkjtuzwtmnflty3767dny"
+      }
+    ],
+    "s": {
+      "/": {
+        "bytes": "7aEDQIscUKVuAIB2Yj6jdX5ru9OcnQLxLutvHPjeMD3pbtHIoErFpo7OoC79Oe2ShgQMLbo2e6dvHh9scqHKEOmieA0"
+      }
+    }
+  }
+}
+```
+
+# 6 Result
+
+A `Result` records the output of the [Task], as well as its success or failure state.
+
+## 6.1 Schema
+
+```ipldsch
+type Result union {
+  | any "ok"
+  | any "error"
+} representation keyed
+}
+```
+
+## 6.2 Variants
+
+## 6.2.1 Success
+
+The success branch MUST contain the value returned from a successful [Task] wrapped in the `"ok"` tag. The exact shape of the returned data is left undefined to allow for flexibility in various Task types.
+
+```json
+{ "ok": 42 }
+```
+
+## 6.2.2 Failure
+
+The failure branch MAY contain detail about why execution failed wrapped in the "error" tag. It is left undefined in this specification to allow for [Task] types to standardize the data that makes sense in their contexts.
+
+If no information is available, this field SHOULD be set to `{}`.
+
+```json
+{
+  "error": {
+    "dev/reason": "unauthorized",
+    "http/status": 401
+  }
+}
+```
+
+## 7 Effect
+
+The result of an [Invocation] MAY include a request for further actions to be performed via "effects". This enables several things: a clean separation of pure return values from reuqesting impure tasks to be performed by the runtime, and gives the runtime the control to decide how (or if!) more work should be performed.
+
+Effects describe requests for future work to be performed. All [Invocation]s in an [Effect] block MUST be treated as concurrent, unless explicit data dependencies between them exist via promise [Pipelines]. The `fx` block contains two fields: `fork` and `join`.
+
+[Task]s listed in the `fork` field are first-class and only ordered by promises; they otherwise SHOULD be considered independent and equal. As such, atomic guarintees such as failure of one effect impling failure of other effects if left undefined.
```

Comment:
```suggestion
[Task]s listed in the `fork` field are first-class and only ordered by promises; they otherwise SHOULD be considered independent and equal. As such, atomic guarantees such as failure of one effect implying failure of other effects if left undefined.
```